test/eip20/eip20.js

```diff
@@ -38,15 +39,15 @@ contract('EIP20', (accounts) => {
     const balanceBefore = await HST.balanceOf.call(accounts[0]);
     assert.strictEqual(balanceBefore.toNumber(), 10000);
 
-    web3.eth.sendTransaction({ from: accounts[0], to: HST.address, value: web3.toWei('10', 'Ether') }, async (err, res) => {
-      expectThrow(new Promise((resolve, reject) => {
-        if (err) reject(err);
+    await assertRevert(new Promise((resolve, reject) => {
```

Comment:
I like using assertRevert on things that return promises, but for things that only take callbacks might it be more legible to just do the check explicitly inside the callback itself? What do you think @maurelian ?

I think this code is right, it's just a little hairy.

Reply:
Agree, I think I've done the same thing in a previous version of the test suite, but it feels like wrapping and unwrapping at the same time.

Reply:
I'm not 100% sure how to do this without bungling up the usage of async/await?

Here's how I understand it: if it's a callback and not using async/await, then it will resolve in its own asynchronous thread, and then the following lines (49+) will execute independently of the sendTransaction. Won't this break the linear asyncs?

Doesn't this need to be chained as usual then with .then() style promises (or more traditional callback hell)?

Is that the intent? I'm not 100% sure I grok how to fix this?

Reply:
Good point. I didn't consider that there's another `await` call made after this. I was able to get this work though: 

    web3.eth.sendTransaction(
      { from: accounts[0], to: HST.address, value: web3.toWei('10', 'Ether') },
      async (err, res) => {
        assert(err);
        assert.strictEqual(res, undefined);

        const balanceAfter = await HST.balanceOf.call(accounts[0]);
        assert.strictEqual(balanceAfter.toNumber(), 10000);
      },
    );

And I think one callback is reasonable. 

Another option could be to promisify it at the top of the file, then you the test itself looks like any other one. 

```
const promiseSend = function (tx) {
  return new Promise((resolve, reject) => {
    web3.eth.sendTransaction(tx, (err, res) => {
      if (err) { reject(err); }
      resolve(res);
    });
  });
};
```

Reply:
Now that I think about it, if we use just the callback, then we have to add a `done` to this function I think. Because it doesn't return a promise. https://mochajs.org/#asynchronous-code

But the second thing you wrote I think would work @maurelian

Reply:
I think I'm okay with merging this as-is though and coming back to this question later.

Reply:
> Now that I think about it, if we use just the callback, then we have to add a done to this function I think. Because it doesn't return a promise. 

FWIW, I had tried that, and got a 'resolution method overspecified' error. The sample I shared was working.